flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -511,6 +514,45 @@ public CompletableFuture<CompletedCheckpoint> triggerCheckpoint(boolean isPeriod
         return triggerCheckpointFromCheckpointThread(checkpointProperties, null, isPeriodic);
     }
 
+    /**
+     * Triggers one new checkpoint with the given checkpointBackupType. If the given
+     * checkpointBackupType is null, then it will fall back to use the CheckpointCoordinator's
+     * checkpointProperties. The return value is a future. It completes when the checkpoint
+     * triggered finishes or an error occurred.
+     *
+     * @param checkpointBackupType specifies the back up type of the checkpoint to trigger.
+     * @return a future to the completed checkpoint.
+     */
+    public CompletableFuture<CompletedCheckpoint> triggerCheckpoint(
+            @Nullable CheckpointBackupType checkpointBackupType) {
+
+        if (checkpointBackupType == null) {
+            return triggerCheckpointFromCheckpointThread(checkpointProperties, null, false);
```

Comment:
might be a little cleaner to avoid calling `trigger...` here. we can just resolve the `checkpointType` and call the `trigger...` at line 543. Also if we directly use `CheckpointType` and removed the `CheckpointBackupType`, this can be even simpler.